libs/SalesforceSDKCore/SalesforceSDKCore/Classes/NativeLoginManager.swift

```diff
@@ -71,4 +71,66 @@ public protocol NativeLoginManager {
     ///
     /// Note: this call will dismiss your login view controller.
     @objc func biometricAuthenticationSuccess()
+    
+    // MARK: Headless, Password-Less Login Via One-Time-Passcode
+    
+    /// Submits a request for a one-time-passcode to the Salesforce headless password-less login flow.
+    /// This fulfills step three of the headless password-less login flow.
+    ///
+    /// See https://help.salesforce.com/s/articleView?id=sf.remoteaccess_headless_passwordless_login_public_clients.htm&type=5
+    ///
+    /// - Parameters:
+    ///   - username: A valid Salesforce username.  Note that email may be used for community users
+    ///   - recaptchaToken: A reCAPTCHA token provided by the reCAPTCHA SDK
+    ///   - otpVerificationMethod: The delivery method for the OTP
+    /// - Returns: An OTP request result with the overall login result and the OTP identifier for
+    /// successful OTP requests
+    ///
+    @objc func submitOtpRequest(
+        username: String,
+        reCaptchaToken: String,
+        otpVerificationMethod: OtpVerificationMethod) async throws -> OtpRequestResult
+    
+    /// Submits a request for a one-time-passcode to the Salesforce headless password-less login flow.
+    /// This fulfills steps eight, eleven and thirteen of the headless password-less login flow.
+    ///
+    /// See https://help.salesforce.com/s/articleView?id=sf.remoteaccess_headless_passwordless_login_public_clients.htm&type=5
+    ///
+    /// - Parameters:
+    ///   - otp: A user-entered OTP
+    ///   - otpIdentifier: The OTP identifier issued by the Headless Identity API
+    ///   - otpVerificationMethod: The OTP verification method used to obtain the OTP identifier
+    /// - Returns: A login result indicating the outcome of the authorization and access token requests
+    ///
+    @objc func submitPasswordlessAuthorizationRequest(
+        otp: String,
+        otpIdentifier: String,
+        otpVerificationMethod: OtpVerificationMethod
+    ) async throws -> NativeLoginResult
+}
+
+/// An Objective-C compatible OTP request result
+@objc(SFOtpRequestResult)
+@objcMembers
+public class OtpRequestResult: NSObject {
```

Comment:
The reason this class exists is so we can pass the `otpIdentifier` back to the view from `submitOtpRequest` so that they can pass it back to us in `submitPasswordlessAuthorizationRequest`?  Couldn't we just hold onto it or am I missing something?

Reply:
That's actually a neat idea but might also have some consequences in scalability.  I usually think of library classes as being more stateless and leaving data and state to the app or test between methods.  In that case, app or test wouldn't have to retain or prime a native login manager instance between calls or tests - they'd always have enough state to let the object go and start over from their own state at any point.

This also lets us skip any null/mutability/synchronization checks on a native login manager property on the second method since the app provides a required value (rather than it being an mutable optional on native login manager).

I could be flexible 'cuz it'd work either way, but those're some reasons I always lean towards returning data to the app/test rather than retaining lots of member variables.

Reply:
I agree it is great to avoid maintaining state in a library whenever possible, but this OTP flow has state inherently just due to waiting for a response from the user.  I suppose it is okay to offload the state management to the app but that also makes consuming the result of the first call more annoying since they get back a separate class that has the result (all the other calls return directly) and on success this value they need to give back to us.  

As you said it will work either way and just seems like a differing of opinion/preference/priority.  I am absolutely _not_ dying on this hill 😄.

Reply:
Also, happy to change it if we want 🏁